### PR TITLE
Bump golang to 1.20.3 and fix golint issues

### PR DESCRIPTION
**Pull request by sethiyash**
- Bump golang to 1.20.3 to get rid of CVE's in go 1.19.6

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -461,7 +461,7 @@ func runYtt(t *testing.T, files testInputFiles, stdinFileName string, flags yttF
 	return string(output)
 }
 
-func runYttExpectingError(t *testing.T, files testInputFiles, stdinFileName string, flags yttFlags, envs []string) string {
```
Comment (praveenrewar) [inline test/e2e/e2e_test.go:464]: I am surprised that this wasn’t being used 😮

**Review by praveenrewar — APPROVED** [review]
LGTM!

**Comment by vmunishwar** [comment]
LGTM - Verified that bumping Go requires golang-lint update to v1.52.2 as well. Reviewed the changes requiring use of blank identifier for unused variables.

**Pull request merged**